aws/resource_aws_securityhub_standards_subscription_test.go

```diff
@@ -0,0 +1,108 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/securityhub"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func testAccAWSSecurityHubStandardsSubscription_basic(t *testing.T) {
+	var standardsSubscription *securityhub.StandardsSubscription
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSSecurityHubStandardsSubscriptionConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSSecurityHubStandardsSubscriptionExists("aws_securityhub_standards_subscription.example", standardsSubscription),
+				),
+			},
+			{
+				ResourceName:      "aws_securityhub_standards_subscription.example",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// Check Destroy - but only target the specific resource (otherwise Security Hub
```

Comment:
I believe this is the correct handling for now. The acceptance testing framework is correctly ordering the destroy operations, but as you mention, since it deletes both the subscription and account resources it'll receive an error with a normal `CheckDestroy` flow:

```
--- FAIL: TestAccAWSSecurityHub (19.59s)
    --- FAIL: TestAccAWSSecurityHub/StandardsSubscription (19.59s)
        --- FAIL: TestAccAWSSecurityHub/StandardsSubscription/basic (19.59s)
            testing.go:599: Error destroying resource! WARNING: Dangling resources
                may exist. The full state and error is shown below.

                Error: Check failed: InvalidAccessException: Account 123456789012 is not subscribed to AWS Security Hub
                	status code: 401, request id: cab84e8c-02f5-11e9-b776-ffce4537012f
```

We should still add `CheckDestroy: testAccCheckAWSSecurityHubAccountDestroy,` to the `TestCase` though, just to be sure we actually have disabled the account at the end of the test. I'll add a note above.